Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Post Item/Feed Post.swift

```diff
@@ -22,59 +26,201 @@ struct FeedPost: View
     @EnvironmentObject var appState: AppState
     
     // arguments
-    let post: APIPostView
+    let postView: APIPostView
     let account: SavedAccount
     
     @Binding var feedType: FeedType
     
     @State private var isShowingSafari: Bool = false
     @State private var isShowingEnlargedImage: Bool = false
     
+    // swipe-to-vote
+    @State var dragPosition: CGSize = .zero
+    @State var prevDragPosition: CGFloat = .zero
+    @State var dragBackground: Color = .systemBackground
+    @State var leftSwipeSymbol: String = "arrow.up"
+    @State var rightSwipeSymbol: String = "arrowshape.turn.up.left"
+    
+    // in-feed reply
+    @State var replyIsPresented: Bool = false
+    @State var replyContents: String = ""
+    @State var replyIsSending: Bool = false
+   
+    
     var body: some View {
-        VStack(spacing: 0) {
-            postItem
-                .contextMenu {
-                    // general-purpose button template for adding more stuff--also nice for debugging :)
-//                        Button {
-//                            print(post)
-//                        } label: {
-//                            Label("Do things", systemImage: "heart")
-//                        }
-                    
-                    // only display share if URL is valid
-                    if let postUrl: URL = URL(string: post.post.apId) {
-                        ShareButton(urlToShare: postUrl, isShowingButtonText: true)
-                    }
+            ZStack {
+                dragBackground
+                HStack(spacing: 0) {
+                    Image(systemName: leftSwipeSymbol)
+                        .font(.title)
+                        .frame(width: 20, height: 20)
+                        .foregroundColor(.white)
+                        .padding(.horizontal, 20)
+                    Spacer()
+                    Image(systemName: rightSwipeSymbol)
+                        .font(.title)
+                        .frame(width: 20, height: 20)
+                        .foregroundColor(.white)
+                        .padding(.horizontal, 20)
                 }
-
-            Divider()
-        }.if (!shouldShowCompactPosts) { viewProxy in
-            viewProxy.padding(.top)
+                postItem
+                    .background(Color.systemBackground)
+                    .offset(x: dragPosition.width)
+                    .highPriorityGesture(
+                        DragGesture(minimumDistance: 15) // min distance prevents conflict with scrolling drag gesture
+                            .onChanged {
+                                let w = $0.translation.width
+                                
+                                if w < -1 * AppConstants.longSwipeDragMin {
+                                    rightSwipeSymbol = "arrowshape.turn.up.left.fill"
```

Comment:
do we need these icons to alter based on the current vote state, so you can swipe to downvote/reset etc?